### PR TITLE
Update Developer Guide on User Interface and Review

### LINKED ISSUE #179: Add implementation of Review class in Developer Guide

### PR DESCRIPTION
**Pull request by Hikoya**
Resolve #179 , #178

### CODE REVIEW

File: docs/diagrams/ArchitectureSequenceDiagram.puml [hunk 1]
```diff
@@ -7,13 +7,13 @@ Participant ":Logic" as logic LOGIC_COLOR
...
-user -[USER_COLOR]> ui : "delete 1"
+user -[USER_COLOR]> ui : "sdel 1"
 activate ui UI_COLOR
 
-ui -[UI_COLOR]> logic : execute("delete 1")
+ui -[UI_COLOR]> logic : execute("sdel 1")
 activate logic LOGIC_COLOR
 
-logic -[LOGIC_COLOR]> model : deletePerson(p)
+logic -[LOGIC_COLOR]> model : deleteStall(p)
```
Comment (clarence-chew) [inline docs/diagrams/ArchitectureSequenceDiagram.puml:16]: `deleteStall(s)` or `deleteStall(stall)` might be preferable here.

File: docs/diagrams/ModelReviewClassDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+UniqueReviewList --> "~* all" Review
+AddressBook *--> "1" UniqueReviewList
+Review *--> Name
+Review *--> Content
+Review *--> Date
+Review *--> Rating
+Review *--> Tag
+
+ModelManager -->"~* filtered" Review
+Review *--> "1" Stall
```
Comment (clarence-chew) [inline docs/diagrams/ModelReviewClassDiagram.puml:26]: The composition diamond doesn't seem right here as when a Review is removed, the corresponding Stall does not need to be removed.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -154,6 +154,33 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+#### **Implementation**
+<img src="images/ModelReviewClassDiagram.png" width="450" />
+
+A `Review`,
+- is stored in `uniqueReviewList` of the Model
+
+A `Review` contains the following attributes,
+1. a `Name`, which represent the name of the Stall associated with the Review
+2. a `Date`, which represent the day, month and year as specified in `DD-MM-YYYY` format
```
Comment (clarence-chew) [inline docs/DeveloperGuide.md:168]: If I remember correctly, the display format was previously agreed to be `DD/MM/YYYY`.

File: docs/diagrams/ModelReviewClassDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Class Content
+Class Date
+Class Rating
+Class Tag
+Class UniqueReviewList
+}
+
+UniqueReviewList --> "~* all" Review
+AddressBook *--> "1" UniqueReviewList
+Review *--> Name
```
Comment (clarence-chew) [inline docs/diagrams/ModelReviewClassDiagram.puml:19]: I think adding multiplicity to these associations may help.

**Review by clarence-chew — COMMENTED** [review]
I think there might be some issues, do review them.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Hikoya) [reply docs/DeveloperGuide.md]: Updated in latest commit

File: docs/diagrams/ModelReviewClassDiagram.puml (hunk 4 above)
Comment (Hikoya) [reply docs/diagrams/ModelReviewClassDiagram.puml]: Updated in latest commit

File: docs/diagrams/ModelReviewClassDiagram.puml (hunk 2 above)
Comment (Hikoya) [reply docs/diagrams/ModelReviewClassDiagram.puml]: Updated in latest commit

File: docs/diagrams/ArchitectureSequenceDiagram.puml (hunk 1 above)
Comment (Hikoya) [reply docs/diagrams/ArchitectureSequenceDiagram.puml]: Updated in latest commit

**Review by MinHeinA — APPROVED** [review]
Issues brought up by @clarence-chew have been resolved. LGTM!

**Pull request merged**